homeassistant/components/sensor/elkm1.py

```diff
@@ -0,0 +1,315 @@
+            ZoneType.FIRE_SUPERVISORY.value: 'fire',
+            ZoneType.KEYFOB.value: 'key',
+            ZoneType.NON_ALARM.value: 'alarm-off',
+            ZoneType.MEDICAL_ALARM.value: 'medical-bag',
+            ZoneType.POLICE_ALARM.value: 'alarm-light',
+            ZoneType.POLICE_NO_INDICATION.value: 'alarm-light',
+            ZoneType.KEY_MOMENTARY_ARM_DISARM.value: 'power',
+            ZoneType.KEY_MOMENTARY_ARM_AWAY.value: 'power',
+            ZoneType.KEY_MOMENTARY_ARM_STAY.value: 'power',
+            ZoneType.KEY_MOMENTARY_DISARM.value: 'power',
+            ZoneType.KEY_ON_OFF.value: 'toggle-switch',
+            ZoneType.MUTE_AUDIBLES.value: 'volume-mute',
+            ZoneType.POWER_SUPERVISORY.value: 'power-plug',
+            ZoneType.TEMPERATURE.value: 'thermometer-lines',
+            ZoneType.ANALOG_ZONE.value: 'speedometer',
+            ZoneType.PHONE_KEY.value: 'phone-classic',
+            ZoneType.INTERCOM_KEY.value: 'deskphone'
+        }
+        return 'mdi:{}'.format(
+            zone_icons.get(self._element.definition, 'alarm-bell'))
+
+    @property
+    def device_state_attributes(self):
+        """Attributes of the sensor."""
+        from elkm1_lib.const import (
+            ZoneLogicalStatus, ZonePhysicalStatus, ZoneType)
+
+        attrs = self.initial_attrs()
+        attrs['physical_status'] = ZonePhysicalStatus(
+            self._element.physical_status).name.lower()
+        attrs['logical_status'] = ZoneLogicalStatus(
+            self._element.logical_status).name.lower()
+        attrs['definition'] = ZoneType(
+            self._element.definition).name.lower()
+        attrs['area'] = self._element.area + 1
+        attrs['bypassed'] = self._element.bypassed
+        attrs['triggered_alarm'] = self._element.triggered_alarm
+        return attrs
+
+    @property
+    def temperature_unit(self):
+        """Return the temperature unit."""
+        from elkm1_lib.const import ZoneType
+        if self._element.definition == ZoneType.TEMPERATURE.value:
+            return self._temperature_unit
+        return None
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        from elkm1_lib.const import ZoneType
+        if self._element.definition == ZoneType.TEMPERATURE.value:
+            return self._temperature_unit
+        if self._element.definition == ZoneType.ANALOG_ZONE.value:
+            return 'V'
+        return None
+
+    def _element_changed(self, element, changeset):
+        from elkm1_lib.const import ZoneLogicalStatus, ZoneType
+        from elkm1_lib.util import pretty_const
+
+        if self._element.definition == ZoneType.TEMPERATURE.value:
+            self._state = temperature_to_state(self._element.temperature, -60)
+        elif self._element.definition == ZoneType.ANALOG_ZONE.value:
+            self._state = self._element.voltage
+        else:
+            self._state = pretty_const(ZoneLogicalStatus(
```

Comment:
What kind of value do we report here? Is it binary, ie true/false or on/off etc?

Reply:
4 values: Normal, Troubled, Violated, Bypassed.

Reply:
Those are some depressing state names. 😂

Since it's four states per entity it fits here as a `Sensor` entity. I was consider binary sensor when I saw "logical".